packages/backend/src/tokens/interstitial.ts

```diff
@@ -67,7 +68,7 @@ export function loadInterstitialFromLocal(
                 try {
                     await Clerk.load({
                         isSatellite: ${isSatellite},
-                        shouldSyncLink: ${isSatellite && debugData?.reason !== 'satellite-cookie-missing'},
+                        shouldSyncLink: ${debugData?.reason === AuthErrorReason.SatelliteCookieNeedsSync},
```

Comment:
🔧 

The change makes sense to me! I'd just like to raise that we probably shouldn't be using `debugData` here, as this is just a bucket of data we assign to the `window.__clerk_debug` var to help us debug some edge cases.

Given that this is a "bucket of data" I expect that we'll end up breaking this check. I think it'd be better if we were more explicit about this.

Reply:
You are correct about not using debugData in that manner. `reason` is also accessible as return value of `authenticateRequest` so I will pass it in a different way here.

@nikosdouvlis Please check my new commit in this PR where I am tackling this.